src/python/include/Simplex_tree_interface.h

```diff
@@ -157,6 +159,33 @@ class Simplex_tree_interface : public Simplex_tree<SimplexTreeOptions> {
     return new_dgm;
   }
 
+  Simplex_tree_interface* collapse_edges(int nb_collapse_iteration) {
+    using Filtered_edge = std::tuple<Vertex_handle, Vertex_handle, Filtration_value>;
+    std::vector<Filtered_edge> edges;
+    for (Simplex_handle sh : Base::skeleton_simplex_range(1)) {
+      if (Base::dimension(sh) == 1) {
+        typename Base::Simplex_vertex_range rg = Base::simplex_vertex_range(sh);
+        std::vector<Vertex_handle> rips_edge(rg.begin(), rg.end());
+        edges.push_back(std::make_tuple(rips_edge[0], rips_edge[1], Base::filtration(sh)));
+      }
+    }
+
+    std::vector<Filtered_edge> remaining_edges;
+    for (int iteration = 0; iteration < nb_collapse_iteration; iteration++) {
+      remaining_edges = Gudhi::collapse::flag_complex_collapse_edges(edges);
+      edges = std::move(remaining_edges);
+      remaining_edges.clear();
+    }
+    Simplex_tree_interface* collapsed_stree_ptr = new Simplex_tree_interface();
+    for (auto remaining_edge : edges) {
+      collapsed_stree_ptr->insert({std::get<0>(remaining_edge)}, 0.);
+      collapsed_stree_ptr->insert({std::get<1>(remaining_edge)}, 0.);
```

Comment:
So vertices are only inserted if they are the extremity of an edge, and always at time 0. I think it would be better to copy the 0-skeleton of the original complex, if we don't want to change the persistence in dimension 0.

Reply:
Collapse edges should copy the 0-skeleton. I also added a test on 39fba06e